mbuild/utils/orderedset.py

```diff
@@ -0,0 +1,42 @@
+"""Ordered set module."""
+from collections.abc import MutableSet
+
+
+class OrderedSet(MutableSet):
+    """An ordered set object with additional convenience methods.
+
+    Taken from code suggested by Vyas in
+    https://github.com/mosdef-hub/mbuild/issues/865
+
+    Methods
+    -------
+    add
+    discard
+    """
+
+    def __init__(self, *args):
```

Comment:
Do we want this constructor to support iterables as well?

Reply:
so right now the constructor can support unpacked iterables, e.g.,
```
mylist = [0,1,2] 
OrderedSet(*mylist)
```
this allows it to also support a sequence of values as args:
```
OrderedSet(0,1,2)
```
I could change it so it assumes it is either being given a sequence of values or an iterable:
```
from collections.abc import Iterable

def __init__(self, *args):
    if isinstance(args[0], Iterable) and not isinstance(args[0], str):
        print("iterable", args[0])
    else: 
        print("sequence of values", args)
```
but that could have some odd behavior if a sequence of iterable arguments were given.
What do you think?

Reply:
what exactly would be needed for that? 

Does [line 18 below this, support that?](https://github.com/mosdef-hub/mbuild/pull/904/files/95f9e52afc40d044f1258f7ab4d15d7b9f65d58a#diff-47e6fa915503bed70a2e9e9cab122a87ae3099cbb6e15b79bd3d494c81416e5eR18)